packages/Meshfree/src/DTK_DetailsMovingLeastSquaresOperatorImpl.hpp

```diff
@@ -0,0 +1,305 @@
+                    target_values( i ) +=
+                        polynomial_coeffs( j ) * source_values( j );
+            } );
+        Kokkos::fence();
+
+        return target_values;
+    }
+
+    static Kokkos::View<Coordinate **, DeviceType> transformSourceCoordinates(
+        Kokkos::View<Coordinate const **, DeviceType> source_points,
+        Kokkos::View<int const *, DeviceType> offset,
+        Kokkos::View<Coordinate const **, DeviceType> target_points )
+    {
+        auto const n_source_points = source_points.extent( 0 );
+        auto const n_target_points = target_points.extent( 0 );
+
+        int const spatial_dim = 3;
+        DTK_REQUIRE( source_points.extent_int( 1 ) == spatial_dim );
+        DTK_REQUIRE( offset.extent( 0 ) == n_target_points + 1 );
+
+        Kokkos::View<Coordinate **, DeviceType> new_source_points(
+            "transformed_source_coords", n_source_points, spatial_dim );
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "transform" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_target_points ),
+            KOKKOS_LAMBDA( const int i ) {
+                for ( int j = offset( i ); j < offset( i + 1 ); j++ )
+                    for ( int k = 0; k < spatial_dim; k++ )
+                        new_source_points( j, k ) =
+                            source_points( j, k ) - target_points( i, k );
+            } );
+
+        return new_source_points;
+    }
+
+    static Kokkos::View<double *, DeviceType>
+    computeRadius( Kokkos::View<Coordinate const **, DeviceType> source_points,
+                   Kokkos::View<int const *, DeviceType> offset )
+    {
+        unsigned int const n_target_points = offset.extent( 0 ) - 1;
+        Kokkos::View<double *, DeviceType> radius( "radius",
+                                                   source_points.extent( 0 ) );
+
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "compute_radius" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_target_points ),
+            KOKKOS_LAMBDA( int const i ) {
+                double distance = 10. * Kokkos::ArithTraits<double>::epsilon();
+                for ( int j = offset( i ); j < offset( i + 1 ); ++j )
+                {
+                    double new_distance = Details::distance(
+                        Point{{source_points( j, 0 ), source_points( j, 1 ),
+                               source_points( j, 2 )}},
+                        Point{{0., 0., 0.}} );
+
+                    if ( new_distance > distance )
+                        distance = new_distance;
+                }
+                // If a point is exactly on the boundary of the compact domain,
+                // its weight will be zero so we need to make sure that no point
+                // is exactly on the boundary.
+                for ( int j = offset( i ); j < offset( i + 1 ); ++j )
+                    radius( j ) = 1.1 * distance;
```

Comment:
How was the parameter of 1.1 selected? Wouldn't 1.01 or smaller be better? If a point is on the radius it effectively has no influence.

Reply:
I think 1.1 is fair for now.  Maybe we can add a token "NOTE" or "FIXME" in the comment above.